pkg/p2p/libp2p/libp2p.go

```diff
@@ -178,6 +178,7 @@ func New(ctx context.Context, o Options) (*Service, error) {
 		peerID := stream.Conn().RemotePeer()
 		i, err := s.handshakeService.Handle(NewStream(stream), peerID)
 		if err != nil {
+			_ = stream.Reset()
```

Comment:
is this necessary?

Reply:
Same as above, if error happened, I don't want to `Close` the stream, but actually `Reset`, which closes both side of the stream.